src/neo/Network/P2P/Payloads/MerkleBlockPayload.cs

```diff
@@ -37,9 +37,9 @@ public static MerkleBlockPayload Create(Block block, BitArray flags)
         public override void Deserialize(BinaryReader reader)
         {
             base.Deserialize(reader);
-            ContentCount = (int)reader.ReadVarInt(int.MaxValue);
-            Hashes = reader.ReadSerializableArray<UInt256>();
-            Flags = reader.ReadVarBytes();
+            ContentCount = (int)reader.ReadVarInt(Block.MaxTransactionsPerBlock);
+            Hashes = reader.ReadSerializableArray<UInt256>(Block.MaxTransactionsPerBlock);
+            Flags = reader.ReadVarBytes(Block.MaxTransactionsPerBlock);
```

Comment:
I'm not sure if this is enough.

Reply:
How about changing it to the following:

```c#
ContentCount = (int)reader.ReadVarInt(Block.MaxTransactionsPerBlock + 1);
Hashes = reader.ReadSerializableArray<UInt256>(Block.MaxTransactionsPerBlock + 1);
Flags = reader.ReadVarBytes(36000);   // 36000 is the limit in FilterLoadPayload.Deserialize
```

Reply:
Flags is related to the number of TX
https://github.com/neo-project/neo/blob/0f75741e2ec471235cba99b6ffc6b38b634dd47e/src/neo/Network/P2P/ProtocolHandler.cs#L236

Reply:
It's a bit array, so I think we should divide it by 8.

Reply:
Why not set `ContentCount = (int)reader.ReadVarInt(Block.MaxTransactionsPerBlock + 1)` and `Hashes = reader.ReadSerializableArray<UInt256>(Block.MaxTransactionsPerBlock + 1)`, as `ContentCount = block.Transactions.Length + 1`

Reply:
Seems good to me too